src/freeze/enable.rs

```diff
@@ -0,0 +1,127 @@
+use super::*;
+
+pub struct EnableFreezeArgs {
+    pub keypair: Option<String>,
+    pub rpc_url: Option<String>,
+    pub cache: String,
+    pub config: String,
+    pub candy_machine: Option<String>,
+    pub freeze_days: Option<u8>,
+}
+
+pub fn process_enable_freeze(args: EnableFreezeArgs) -> Result<()> {
+    let sugar_config = sugar_setup(args.keypair.clone(), args.rpc_url.clone())?;
+    let client = setup_client(&sugar_config)?;
+    let program = client.program(CANDY_MACHINE_ID);
+    let config_data = get_config_data(&args.config)?;
+
+    // The candy machine id specified takes precedence over the one from the cache.
+    let candy_machine_id = match args.candy_machine {
+        Some(ref candy_machine_id) => candy_machine_id.to_owned(),
+        None => {
+            let cache = load_cache(&args.cache, false)?;
+            cache.program.candy_machine
+        }
+    };
+
+    let candy_pubkey = Pubkey::from_str(&candy_machine_id)
+        .map_err(|_| anyhow!("Failed to parse candy machine id: {}", &candy_machine_id))?;
+
+    // Freeze days specified takes precedence over the one from the config.
+    let freeze_time = if let Some(freeze_days) = args.freeze_days {
+        (freeze_days * 24 * 60 * 60) as i64
```

Comment:
This line generated a panic for me:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/729235/190191460-3253b305-9d01-4fc8-a1b3-a0036b30d193.png">
My guess is that it was trying to do the multiplication as `u8`. I changed to `(freeze_days as i64) * 24 * 60 * 60` and it worked.

Reply:
[Fixed](https://github.com/metaplex-foundation/sugar/pull/345/commits/566258c4317e8301d636b1c47eead7d2dc8ac4e6).

![Selection_961](https://user-images.githubusercontent.com/1684605/190212865-add16228-c951-4157-90b5-0000fcd21900.png)